### PR TITLE
style the invite pages

### PR DESCRIPTION
**Pull request by staltz**
# Desktop

## Invite created

invite-created-desktop

## Join

join-default-desktop

## Join fallback

join-fallback-desktop

## Join manually

join-manually-desktop

## Join manually succeeded

joined-desktop

<details>
<summary>Mobile</summary>

## Invite created

invite-created-mobile

## Join

join-default-mobile

## Join fallback

join-fallback-mobile

## Join manually

join-manually-mobile

## Join manually succeeded

joined-mobile

</details>

### CODE REVIEW

**Comment by staltz** [comment]
Alright, I'll review the tests that I broke completely.

**Comment by staltz** [comment]
@cryptix This is ready for review :)

File: web/handlers/invites.go [hunk 1]
```diff
@@ -60,13 +54,88 @@ func (h inviteHandler) presentFacade(rw http.ResponseWriter, req *http.Request)
 
 	joinRoomURI.RawQuery = queryVals.Encode()
 
+	return joinRoomURI.String()
```
Comment (cryptix) [inline web/handlers/invites.go:57]:
```suggestion
	return template.URL(joinRoomURI.String())
```

**Review by cryptix — APPROVED** [review]
Small nitpick on the `buildJoinRoomURI` function but looks all good from my end.

I just eyeballed the template changes, tho.

Big +1 for the docs folder!

**Comment by staltz** [comment]
@cryptix Thanks for the suggested changes, I incorporated them now. Also, I added the chart.graphml file, which is the source file in case we need to update that later.

**Comment by cryptix** [comment]
Ah... that's what you get for making code changes on github..


> web/handlers/invites.go:82:27: cannot use joinRoomURI (type "html/template".URL) as type string in argument to qrcode.New

https://github.com/ssb-ngi-pointer/go-ssb-room/runs/2277216659#step:6:7

**Comment by staltz** [comment]
Yeah I was looking into now, not sure how to get a string from that.

File: web/handlers/invites.go [hunk 2]
```diff
@@ -60,13 +54,88 @@ func (h inviteHandler) presentFacade(rw http.ResponseWriter, req *http.Request)
...
+		return nil, fmt.Errorf("failed to find room's description: %w", err)
+	}
+
+	joinRoomURI := buildJoinRoomURI(h, token)
+
+	urlTo := web.NewURLTo(router.CompleteApp())
+	fallbackURL := urlTo(router.CompleteInviteFacadeFallback, "token", token)
+
+	// generate a QR code with the token inside so that you can open it easily in a supporting mobile app
+	qrCode, err := qrcode.New(joinRoomURI, qrcode.Medium)
```
Comment (cryptix) [inline web/handlers/invites.go:82]:
```suggestion
	qrCode, err := qrcode.New(string(joinRoomURI), qrcode.Medium)
````

this should do the trick.

**Comment by cryptix** [comment]
> not sure how to get a string from that.

In the case where it's a _type aliases_ (i.e. `type Foo int`) you can do this:

```go
type Foo int

var i = 42
var asF = Foo(i)

var asIntAgain = int(asF)
```

Same for `bool` or `string`. It's just for complex things like structs where it get's hairy but the compiler will yell at you also if you'd try.

**Comment by staltz** [comment]
Thanks! I'll let you press the big green button

**Pull request merged**